### PR TITLE
Add support for GeoIPLite2-ASN database

### LINKED ISSUE #116: GeoLite2-ASN Database doesn't work with 4.1.1 release.
I am testing release 4.1.1 of the filter using both the `GeoIP2-ISP.mmdb` and `GeoLite2-ASN.mmdb` databases. The updated filter was added to Logstash using `logstash-plugin update logstash-filter-geoip`

The good news is that the GeoIP2-ISP database works great. This database was already supported in `com.maxmind.geoip2` 2.8.0 and `com.maxmind.db` 1.2.1, so it is expected that it should work once the functionality was added to the filter. The output from this database looks like this...

```
"geoip_asn_dst" => {
    "as_org" => "Vodafone Kabel Deutschland GmbH",
    "ip" => "123.123.123.123",
    "isp" => "Vodafone Kabel Deutschland",
    "organization" => "Vodafone Kabel Deutschland",
    "asn" => 31334
}
```

The GeoLite2-ASN database however does not work. I suspect this is due to the fact that support for this database was not included in the MaxMind libraries until `com.maxmind.geoip2` 2.9.0 and `com.maxmind.db` 1.2.2.

Notice that in `lib/logstash-filter-geoip_jars.rb` the old MaxMind libraries are still referenced, which may be the cause of the problem...

```
require_jar('com.maxmind.geoip2', 'geoip2', '2.8.0')
require_jar('com.maxmind.db', 'maxmind-db', '1.2.1')
```

This is where my Java and Logstash plugin development knowledge ends, so I haven't troubleshot the issue further. However since the plugin update installed only the old MaxMind libraries I strongly suspect this is the issue.

For completeness, here is the error seen when using the GeoLite2-ASN database.

```
[2017-05-20T12:16:30,878][ERROR][logstash.pipeline        ] Exception in pipelineworker, the pipeline stopped processing new events, please check your filter configuration and restart Logstash. {"exception"=>"Invalid attempt to open a GeoLite2-ASN database using the isp method", "backtrace"=>["com.maxmind.geoip2.DatabaseReader.get(com/maxmind/geoip2/DatabaseReader.java:146)", "com.maxmind.geoip2.DatabaseReader.isp(com/maxmind/geoip2/DatabaseReader.java:275)", "org.logstash.filters.GeoIPFilter.retrieveIspGeoData(org/logstash/filters/GeoIPFilter.java:317)", "org.logstash.filters.GeoIPFilter.handleEvent(org/logstash/filters/GeoIPFilter.java:130)", "java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:498)", "RUBY.filter(/Users/rob/local/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-geoip-4.1.1-java/lib/logstash/filters/geoip.rb:122)", "LogStash::Filters::Base.do_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:145)", "LogStash::Filters::Base.do_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:145)", "LogStash::Filters::Base.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:164)", "LogStash::Filters::Base.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:164)", "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)", "LogStash::Filters::Base.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:161)", "LogStash::Filters::Base.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:161)", "LogStash::FilterDelegator.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filter_delegator.rb:43)", "LogStash::FilterDelegator.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filter_delegator.rb:43)", "RUBY.initialize((eval):499931)", "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)", "RUBY.initialize((eval):499926)", "org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)", "RUBY.initialize((eval):499947)", "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)", "RUBY.initialize((eval):499942)", "org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)", "RUBY.initialize((eval):500009)", "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)", "RUBY.initialize((eval):500005)", "org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)", "RUBY.initialize((eval):500685)", "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)", "RUBY.initialize((eval):500680)", "org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)", "RUBY.initialize((eval):500720)", "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)", "RUBY.initialize((eval):500706)", "org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)", "RUBY.filter_func((eval):22418)", "LogStash::Pipeline.filter_batch(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:370)", "LogStash::Pipeline.filter_batch(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:370)", "org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)", "LogStash::Util::WrappedSynchronousQueue::ReadBatch.each(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:224)", "LogStash::Util::WrappedSynchronousQueue::ReadBatch.each(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:224)", "org.jruby.RubyHash.each(org/jruby/RubyHash.java:1342)", "LogStash::Util::WrappedSynchronousQueue::ReadBatch.each(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:223)", "LogStash::Util::WrappedSynchronousQueue::ReadBatch.each(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:223)", "LogStash::Pipeline.filter_batch(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:369)", "LogStash::Pipeline.filter_batch(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:369)", "RUBY.worker_loop(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:350)", "RUBY.start_workers(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:317)", "java.lang.Thread.run(java/lang/Thread.java:745)"]}
Exception in thread "[main]>worker7" java.lang.UnsupportedOperationException: Invalid attempt to open a GeoLite2-ASN database using the isp method
	at com.maxmind.geoip2.DatabaseReader.get(com/maxmind/geoip2/DatabaseReader.java:146)
	at com.maxmind.geoip2.DatabaseReader.isp(com/maxmind/geoip2/DatabaseReader.java:275)
	at org.logstash.filters.GeoIPFilter.retrieveIspGeoData(org/logstash/filters/GeoIPFilter.java:317)
	at org.logstash.filters.GeoIPFilter.handleEvent(org/logstash/filters/GeoIPFilter.java:130)
	at java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:498)
	at RUBY.filter(/Users/rob/local/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-geoip-4.1.1-java/lib/logstash/filters/geoip.rb:122)
	at LogStash::Filters::Base.do_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:145)
	at LogStash::Filters::Base.do_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:145)
	at LogStash::Filters::Base.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:164)
	at LogStash::Filters::Base.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:164)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)
	at LogStash::Filters::Base.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:161)
	at LogStash::Filters::Base.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:161)
	at LogStash::FilterDelegator.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filter_delegator.rb:43)
	at LogStash::FilterDelegator.multi_filter(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/filter_delegator.rb:43)
	at RUBY.initialize((eval):499931)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)
	at RUBY.initialize((eval):499926)
	at org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)
	at RUBY.initialize((eval):499947)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)
	at RUBY.initialize((eval):499942)
	at org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)
	at RUBY.initialize((eval):500009)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)
	at RUBY.initialize((eval):500005)
	at org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)
	at RUBY.initialize((eval):500685)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)
	at RUBY.initialize((eval):500680)
	at org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)
	at RUBY.initialize((eval):500720)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)
	at RUBY.initialize((eval):500706)
	at org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)
	at RUBY.filter_func((eval):22418)
	at LogStash::Pipeline.filter_batch(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:370)
	at LogStash::Pipeline.filter_batch(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:370)
	at org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)
	at LogStash::Util::WrappedSynchronousQueue::ReadBatch.each(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:224)
	at LogStash::Util::WrappedSynchronousQueue::ReadBatch.each(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:224)
	at org.jruby.RubyHash.each(org/jruby/RubyHash.java:1342)
	at LogStash::Util::WrappedSynchronousQueue::ReadBatch.each(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:223)
	at LogStash::Util::WrappedSynchronousQueue::ReadBatch.each(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:223)
	at LogStash::Pipeline.filter_batch(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:369)
	at LogStash::Pipeline.filter_batch(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:369)
	at RUBY.worker_loop(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:350)
	at RUBY.start_workers(/Users/rob/local/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:317)
	at java.lang.Thread.run(java/lang/Thread.java:745)
```

### PR DESCRIPTION
**Pull request by suyograo**
Fixes #116

Supporting the GeoIPLite2-ASN requires new Java dependencies from MaxMind.

### CODE REVIEW

File: ci/run.sh [hunk 1]
```diff
@@ -1,6 +1,14 @@
+#!/bin/bash
+current_dir="$(dirname "$0")"
+
 bundle install
 bundle exec rake gradle.properties
 ./gradlew assemble
 bundle exec rake vendor
+
+# we don't want to bundle ASN DB with this gem, so we download this just for tests.
+curl -sL http://geolite.maxmind.com/download/geoip/database/GeoLite2-ASN.tar.gz > GeoLite2-ASN.tar.gz
```
Comment (original-brownbear) [inline ci/run.sh]: @suyograo you could handle the downloading and extracting via Gradle like I did for the user agent parser https://github.com/logstash-plugins/logstash-filter-useragent/blob/master/build.gradle#L65

Would be a little more portable and less hacky than `dirname` + `curl`.

**Comment by original-brownbear** [comment]
@suyograo same here LGTM :) just a very minor suggestion.

**Comment by suyograo** [comment]
@original-brownbear handled in https://github.com/logstash-plugins/logstash-filter-geoip/pull/120/commits/016bef57cee9e5be0ca537f084d9b3e7dbfdd054

**Comment by original-brownbear** [comment]
@suyograo LGTM :)

**Pull request merged**